community/kernel/src/main/java/org/neo4j/kernel/impl/store/NodeStore.java

```diff
@@ -212,6 +212,46 @@ public NodeRecord loadRecord( long id, NodeRecord record )
         }
     }
 
+    public boolean loadRecord( long id, NodeRecord record, PageCursor cursor)
+    {
+        long pageId = pageIdForRecord( id );
+        int offset = offsetForId( id );
+
+        try
+        {
+            cursor.next( pageId );
+
+            boolean isInUse = false;
+            if ( cursor.next() )
```

Comment:
You are positioning it with the `cursor.net( pageId )` and then you move to `pageId + 1` with the `cursor.next()` call in the if-clause. Is that really how it's supposed to be?